cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -1144,10 +1151,27 @@ private void setMapOutputClassesIfNeeded(Job job, @Nullable TypeToken<?> mapperT
     }
   }
 
+  // Regex pattern for configuration source if it is set programmatically. This constant is not defined in Hadoop
```

Comment:
Note I moved this back here. Originally moved it to common because I was using it in dataset code, but then all my changes moved back to this class, so it is fine to be here. However, I had to fix it to look at the first, not the last source of the property.